src/main/java/NestedNullRefactoring.java

```diff
@@ -0,0 +1,155 @@
+import java.lang.reflect.Modifier;
+import java.util.Dictionary;
+import java.util.Hashtable;
+import java.util.List;
+
+import org.eclipse.jdt.core.dom.AST;
+import org.eclipse.jdt.core.dom.ASTNode;
+import org.eclipse.jdt.core.dom.Block;
+import org.eclipse.jdt.core.dom.Expression;
+import org.eclipse.jdt.core.dom.IMethodBinding;
+import org.eclipse.jdt.core.dom.InfixExpression;
+import org.eclipse.jdt.core.dom.MethodDeclaration;
+import org.eclipse.jdt.core.dom.MethodInvocation;
+import org.eclipse.jdt.core.dom.NullLiteral;
+import org.eclipse.jdt.core.dom.ParenthesizedExpression;
+import org.eclipse.jdt.core.dom.PrefixExpression;
+import org.eclipse.jdt.core.dom.PrimitiveType;
+import org.eclipse.jdt.core.dom.ReturnStatement;
+import org.eclipse.jdt.core.dom.SimpleName;
+import org.eclipse.jdt.core.dom.Statement;
+import org.eclipse.jdt.core.dom.Type;
+import org.eclipse.jdt.core.dom.rewrite.ASTRewrite;
+
+/**
+ * This class represents a refactoring in which invocations to private one line
+ * methods which return the nullness of a value are replaced with the method's
+ * one-line null check. Preserves semantics by copying the invoked code
+ * precisely.
+ */
+public class NestedNullRefactoring extends Refactoring {
+	public static final String NAME = "NestedNullRefactoring";
+
+	private final Dictionary<IMethodBinding, Expression> applicableMethods;
+
+	public NestedNullRefactoring() {
+		applicableMethods = new Hashtable<>();
+	}
+
+	@Override
+	public boolean isApplicable(ASTNode node) {
+		if (node instanceof MethodInvocation invocation) {
+			return isApplicable(invocation);
+		}
+
+		if (node instanceof MethodDeclaration declaration) {
+			return isApplicable(declaration);
+		}
+
+		return false;
+	}
+
+	/*
+	 * Returns true iff the provided invocation is of a registered one-line method
+	 * that returns the result of a null check
+	 */
+	public boolean isApplicable(MethodInvocation invocation) {
+		if (applicableMethods.get(invocation.resolveMethodBinding()) != null) {
+			System.out.println("[DEBUG] Invocation of applicable method found");
+			return true;
+		}
+		return false;
+	}
+
+	/*
+	 * Returns true iff Node is a one-line private or package-private method that
+	 * returns the result of a null check
+	 */
+	public boolean isApplicable(MethodDeclaration declaration) {
+		Type retType = declaration.getReturnType2();
```

Comment:
Why `getReturnType2`? When a method has such an unusual name, there's usually a reason. Why are we calling the version with the odd name instead of `getReturnType`? This is a good example of a place where a comment explaining what's going on is appropriate: the code itself is confusing.